### PR TITLE
Fix/eth flow

### PR DESCRIPTION
**Pull request by rodrigopavezi**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "artizen-frontend",
-  "version": "0.1.23",
+  "version": "0.1.24",
```
Comment (rubelux) [inline package.json:3]: nice!

File: src/components/Donations/ProcessCrypto/index.tsx [hunk 2]
```diff
@@ -5,10 +5,10 @@ import { breakpoint, typography, palette } from '@theme'
 import { useEffect } from 'react'
 
 const ProcessCrypto = () => {
-  const { cryptoStage, setCryptoStage, donationMethod, error, restart, retry } = useProcessDonation()
+  const { cryptoStage, setCryptoStage, donationMethod, error, restart, retry, swapId } = useProcessDonation()
 
   useEffect(() => {
-    setCryptoStage?.(donationMethod !== 'ethereum' ? 'building' : 'swapping')
+    setCryptoStage?.(donationMethod !== 'ethereum' ? 'building' : swapId ? 'bridging' : 'swapping')
```
Comment (rubelux) [inline src/components/Donations/ProcessCrypto/index.tsx:11]: nice

File: src/lib/processDonation.tsx [hunk 3]
```diff
@@ -286,23 +282,25 @@ export const ProcessDonationProvider = ({ children, chains }: IProcessDonationPr
...
-          console.error('Ignoring message from courier', message.title)
-          break
-      }
-    })
-  }, [])
+    if (metadata?.publicAddress && amount) {
+      courier.transport.intercept((message: ICourierMessage) => {
+        switch (message.title) {
+          case 'Payment is COMPLETE':
+            handleTopUpComplete()
```
Comment (rubelux) [inline src/lib/processDonation.tsx:289]: can you pass through the dependencies as agreed this afternoon?  Do it for 3 functions

File: src/lib/processDonation.tsx [hunk 4]
```diff
@@ -317,14 +315,14 @@ export const ProcessDonationProvider = ({ children, chains }: IProcessDonationPr
 
   useEffect(() => {
     if (fundsTransfered && fundsTransfered.statusCode === 2) {
-      createTopUpOrder(fundsTransfered.exitHash, swapId)
+      createTopUpOrder(fundsTransfered.exitHash, swapId, formatUSDC(fundsTransfered.exitAmount))
     }
   }, [fundsTransfered])
 
   const handleBuildingDonationRestart = async () => {
```
Comment (rubelux) [inline src/lib/processDonation.tsx:322]: pass metadata?.```publicAddress!```, and ```amount``` attibutes

**Review by rubelux — CHANGES_REQUESTED** [review]

File: src/lib/processDonation.tsx (hunk 3 above)
Comment (rodrigopavezi) [reply src/lib/processDonation.tsx:289]: Sorry @rubelux why would I pass to them if they don't use them at all?

**Pull request merged**